pages/coinbase/homepage.py

```diff
@@ -1,45 +1,237 @@
 class HomePage:
 
-    def __init__(self, driver):
+    def __init__(self, driver, dismissCookies=True):
 
         driver.get(config.coinbase_domain)
-        self.driver = driver
+        cookie_dismiss_button = DialogueBox(driver,self.cookie_dismiss_button_loc,self.visible_dialogue_box_loc)        
+        if dismissCookies: cookie_dismiss_button.click()
 
+        self.driver = driver
         self.build_elements()
 
-    # NAVIGATION BAR
-
-    home_link_loc =         (by.XPATH, "//a[@title='Home']")
-    prices_link_loc =       (by.XPATH, "//a[@title='Prices']")
-    products_menu_loc =     (by.XPATH, "//button[contains(text(),'Products')]")
-    company_menu_loc =      (by.XPATH, "//button[contains(text(),'Company')]")
-    earn_crypto_link_loc =  (by.XPATH, "//a[@title='Earn crypto']")
-    sign_in_link_loc =      (by.XPATH, "//a[@title='Sign in']")
-    header_get_started_button_loc =(
-        by.XPATH, "//a[.='Get started']")
-
-    def build_elements(self):
-
-        # NAVIGATION BAR
-
-        self.home_link =        Link(self.driver, self.home_link_loc)
-        self.prices_link =      Link(self.driver, self.prices_link_loc)
-        self.products_menu =    ReactiveMenu(self.driver,self.products_menu_loc)
-        self.company_menu =     ReactiveMenu(self.driver, self.company_menu_loc)
-        self.earn_crypto_link = Link(self.driver, self.earn_crypto_link_loc)
-        self.sign_in_link =     Link(self.driver, self.sign_in_link_loc)
-        self.header_get_started_button = Button(
-            self.driver, self.header_get_started_button_loc)
+    # A dismiss cookie dialogue box appears and hides elements, clicking
+    # the dismiss button performed in the init method fixes this issue  
+    visible_dialogue_box_loc =      (by.XPATH, "//div[@class='sc-fzoLsD gkNzOD']")
+    cookie_dismiss_button_loc =     (by.XPATH, "//button[.='Dismiss']")
 
+    # NAVIGATION BAR
 
+    home_link_loc =                 (by.XPATH, "//a[@title='Home']")
+    prices_link_loc =               (by.XPATH, "//a[@title='Prices']")
+    products_menu_loc =             (by.XPATH, "//button[contains(text(),'Products')]")
+    company_menu_loc =              (by.XPATH, "//button[contains(text(),'Company')]")
+    earn_crypto_link_loc =          (by.XPATH, "//a[@title='Earn crypto']")
+    sign_in_link_loc =              (by.XPATH, "//a[@title='Sign in']")
+    header_get_started_button_loc = (by.XPATH, "//a[.='Get started']")
+
+         
+    # CREATE ACCOUNT
+
+    get_started_email_field_loc =   (by.XPATH, "//input[@type='email']")
+    get_started_button_loc =        (by.XPATH, "//input[@type='email']/following-sibling::button")
+
+    # CRYPTO CURRENCY MENU
+
+    bitcoin_text_link_loc =         (by.XPATH, "//h4[.='Bitcoin']")
+    bitcoin_BTC_link_loc =          (by.XPATH, "//h4[.='BTC']")
+    bitcoin_logo_link_loc =         (by.XPATH, "//img[@alt='Bitcoin logo']")
+    bitcoin_price_button_loc =      (by.XPATH, "//a[@href='/price/bitcoin']/button")
```

Comment:
For testing purposes I'm not a huge fan of finding buttons by @href. This can go wrong in too many ways depending on the direction of the wind and what brand of coffee the dev drank. Additionally, one of the things we might test is that the buttons go to the right place, and if someone mixed up the @href of two buttons your xcode will just end up finding the other button and clicking it, and going to the link you expected it to go to, and the test would pass even though it should fail.

THAT BEING SAID, you _should_ be able to find this button using the great grandparent element of this one, `//tr[contains(@class,'AssetTableRow')][1]`, but that's terrible xpath as well. In this case I would ask a dev to put an @id in here, but we can't do that. so..... 

Don't worry about it. Your xpath here is fine. Good job using what you had, it's probably what I would have done. I just wanted to mention that under normal circumstances (i.e., we can talk to the devs) we wouldn't do this.